### PR TITLE
Use code snippets for code examples

### LINKED ISSUE #31: Link code snippets from existing examples in KUDO
**What would you like to be added**:
Currently, the example YAMLs in, e.g., [Developing Operators](https://kudo.dev/docs/developing-operators.html) are part of the documentation. To always ensure their correctness, they should be [code snippets](https://v1.vuepress.vuejs.org/guide/markdown.html#import-code-snippets) taken from the examples [provided by KUDO](https://github.com/kudobuilder/kudo/tree/master/config/samples) instead. This could be done by adding a pinned version of KUDO as a Git submodule to this repository.

**Why is this needed**:
This will ensure that the examples in the documentation work. Especially once their correctness is ensured with integration tests (e.g., https://github.com/kudobuilder/kudo/pull/828).

### PR DESCRIPTION
**Pull request by nfnt**
**What this PR does / why we need it**:
To ensure that examples are in sync with KUDO, their code is taken directly from the KUDO repository. The KUDO repository is added as a submodule. The [markdown-it-vuepress-code-snippet-enhanced](https://github.com/fabbballin/markdown-it-vuepress-code-snippet-enhanced) plugin is used for this, because it supports viewing only a range of lines of a snippet.

**Which issue(s) this PR fixes**:

Fixes #31 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by nfnt** [comment]
~WIP, need to double-check that I'm getting the submodule stuff right.~

File: content/docs/developing-operators.md [hunk 1]
```diff
@@ -136,22 +76,11 @@ Plan foo
 
 Plans consists of one or more `phases`. `Phases` consists of one or more `steps`. `Steps` contain one or more `tasks` (those are defined in the section we talked about in the last paragraph). Both phases and also steps can be configured with an execution `strategy`, either `serial` or `parallel`.
 
-The sample has a `deploy` plan with a `deploy-phase` and a `deploy-step`. From the `deploy-step` the `deploy-task` is referenced. This task gets executed when an instance is created using the operator.
+The sample has a `deploy` plan with a `main` and a `everything`. From `everything` the `app` is referenced. This task gets executed when an instance is created using the operator.
```
Comment (meichstedt) [inline content/docs/developing-operators.md]:
```suggestion
The sample has a `deploy` plan with a phase `main` and a step `everything`. From `everything` the `app` is referenced. This task gets executed when an instance is created using the operator.
```

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "kudo"]
+	path = kudo
+	url = https://github.com/kudobuilder/kudo.git
```
Comment (meichstedt) [inline .gitmodules:3]: Are we OK with master or would we want to pin a specific branch/tag?

File: content/docs/developing-operators.md (hunk 1 above)
Comment (meichstedt) [reply content/docs/developing-operators.md]: One downside of including files is that someone might change IDs in the file which leads to weird documentation. I don't think we can prevent this though and the benefit of including them weighs heavier I think.

**Review by meichstedt — APPROVED** [review]
Thanks @nfnt, that's pretty neat!

Do you know if the build will fail in case a file was moved in the kudo repo? I hope it does :)

File: .gitmodules (hunk 2 above)
Comment (nfnt) [reply .gitmodules:3]: It's pinned to `v0.6.0` which has a SHA of `c74d22743d6ec37e24e89eaeb64803597187fe08`.

**Comment by nfnt** [comment]
Good point to fail the build if a file is removed. Let me check this.

**Comment by nfnt** [comment]
The build doesn't fail if a wrong file name is included. The rendered html shows a "File not found". I'll take a look if this can get checked somehow.

**Comment by nfnt** [comment]
Looks like this is a limitation in Vuepress. Even [regular code snippet imports](https://vuepress.vuejs.org/guide/markdown.html#import-code-snippets) only create a "File not found" message but don't fail the render.

**Comment by nfnt** [comment]
Created upstream issue https://github.com/vuejs/vuepress/issues/1872.

**Review by kensipe — APPROVED** [review]

**Review by meichstedt — APPROVED** [review]
`yarn docs:build` actually fails when sources aren't present:

```bash
error Code snippet path not found: /Users/matthias/Documents/projects/kudobuilder/www/kudo/config/samples/first-operator/operator.yaml
error Code snippet path not found: /Users/matthias/Documents/projects/kudobuilder/www/kudo/config/samples/first-operator/templates/deployment.yaml
error Code snippet path not found: /Users/matthias/Documents/projects/kudobuilder/www/kudo/config/samples/first-operator/params.yaml
Language does not exist text
wait Rendering static HTML...
wait vuepress-plugin-feed Adding pages/posts as feed items...
success vuepress-plugin-feed added 7 page(s) as feed item(s)
wait vuepress-plugin-feed Checking feeds that need to be generated...
success vuepress-plugin-feed rss2 feed file generated and saved to content/.vuepress/dist/rss.xml
success vuepress-plugin-feed atom1 feed file generated and saved to content/.vuepress/dist/feed.atom
success vuepress-plugin-feed json1 feed file generated and saved to content/.vuepress/dist/feed.json
success Generated static files in content/.vuepress/dist.

error Command failed with exit code 1.
```

**Comment by alenkacz** [comment]
@nfnt should we close this or is it still relevant?

**Comment by nfnt** [comment]
This is still relevant, but I'll have to update this to KUDO 0.8.0. There's an ongoing discussion with @meichstedt about having a workflow that does all this once new versions of KUDO are released. Because we're using the code of a tagged KUDO version, this tag needs to changed for each version bump of KUDO. I'll have to document this.

**Review by meichstedt — APPROVED** [review]
Thanks for documenting the submodule approach @nfnt!

**Review by meichstedt — APPROVED** [review]

**Pull request merged**